Mathlib/Topology/ContinuousFunction/Bounded.lean

```diff
@@ -0,0 +1,1627 @@
+#align bounded_continuous_function.mul_apply BoundedContinuousFunction.mul_apply
+
+instance : NonUnitalRing (α →ᵇ R) :=
+  FunLike.coe_injective.nonUnitalRing _ coe_zero coe_add coe_mul coe_neg coe_sub
+    (fun _ _ => coe_nsmul _ _) fun _ _ => coe_zsmul _ _
+
+instance nonUnitalSeminormedRing : NonUnitalSeminormedRing (α →ᵇ R) :=
+  { BoundedContinuousFunction.seminormedAddCommGroup with
+    norm_mul := fun _ _ =>
+      norm_ofNormedAddCommGroup_le _ (mul_nonneg (norm_nonneg _) (norm_nonneg _)) _
+    -- Porting note: These 5 fields were missing. Add them.
+    left_distrib, right_distrib, zero_mul, mul_zero, mul_assoc }
+
+end Seminormed
+
+instance nonUnitalNormedRing [NonUnitalNormedRing R] : NonUnitalNormedRing (α →ᵇ R) :=
+  { BoundedContinuousFunction.nonUnitalSeminormedRing,
+    BoundedContinuousFunction.normedAddCommGroup with }
+
+end NonUnital
+
+section Seminormed
+
+variable [SeminormedRing R]
+
+@[simp]
+theorem coe_npowRec (f : α →ᵇ R) : ∀ n, ⇑(npowRec n f) = (⇑f) ^ n
+  | 0 => by rw [npowRec, pow_zero, coe_one]
+  | n + 1 => by rw [npowRec, pow_succ, coe_mul, coe_npowRec f n]
+#align bounded_continuous_function.coe_npow_rec BoundedContinuousFunction.coe_npowRec
+
+instance hasNatPow : Pow (α →ᵇ R) ℕ where
+  pow f n :=
+    { toContinuousMap := f.toContinuousMap ^ n
+      map_bounded' := by simpa [coe_npowRec] using (npowRec n f).map_bounded' }
+#align bounded_continuous_function.has_nat_pow BoundedContinuousFunction.hasNatPow
+
+@[simp]
+theorem coe_pow (n : ℕ) (f : α →ᵇ R) : ⇑(f ^ n) = (⇑f) ^ n := rfl
+#align bounded_continuous_function.coe_pow BoundedContinuousFunction.coe_pow
+
+@[simp]
+theorem pow_apply (n : ℕ) (f : α →ᵇ R) (v : α) : (f ^ n) v = f v ^ n := rfl
+#align bounded_continuous_function.pow_apply BoundedContinuousFunction.pow_apply
+
+instance : NatCast (α →ᵇ R) :=
+  ⟨fun n => BoundedContinuousFunction.const _ n⟩
+
+@[simp, norm_cast]
+theorem coe_natCast (n : ℕ) : ((n : α →ᵇ R) : α → R) = n := rfl
+#align bounded_continuous_function.coe_nat_cast BoundedContinuousFunction.coe_natCast
+
+instance : IntCast (α →ᵇ R) :=
+  ⟨fun n => BoundedContinuousFunction.const _ n⟩
+
+@[simp, norm_cast]
+theorem coe_intCast (n : ℤ) : ((n : α →ᵇ R) : α → R) = n := rfl
+#align bounded_continuous_function.coe_int_cast BoundedContinuousFunction.coe_intCast
+
+instance ring : Ring (α →ᵇ R) :=
+  FunLike.coe_injective.ring _ coe_zero coe_one coe_add coe_mul coe_neg coe_sub
+    (fun _ _ => coe_nsmul _ _) (fun _ _ => coe_zsmul _ _) (fun _ _ => coe_pow _ _) coe_natCast
+    coe_intCast
+
+instance : SeminormedRing (α →ᵇ R) :=
+  { show Ring (α →ᵇ R) from ring,  -- porting note: this was not present in the original
+    BoundedContinuousFunction.nonUnitalSeminormedRing with }
+
+end Seminormed
+
+instance [NormedRing R] : NormedRing (α →ᵇ R) :=
+  { show Ring (α →ᵇ R) from ring,  -- porting note: this was not present in the original
+    BoundedContinuousFunction.nonUnitalNormedRing with }
```

Comment:
```suggestion
instance : SeminormedRing (α →ᵇ R) :=
  { show Ring (α →ᵇ R) from inferInstance,  -- porting note: this was not present in the original
    BoundedContinuousFunction.nonUnitalSeminormedRing with }

end Seminormed

instance [NormedRing R] : NormedRing (α →ᵇ R) :=
  { show Ring (α →ᵇ R) from inferInstance,  -- porting note: this was not present in the original
    BoundedContinuousFunction.nonUnitalNormedRing with }
```
This works, so Lean does find the instance. I guess it just doesn't try to fill in the missing fields by type class inference anymore without being explicitly told to do so.